hummingbot/market/dolomite/dolomite_market

```diff
@@ -0,0 +1,1362 @@
+            InFlightOrder typed_in_flight_order
+
+        for in_flight_order in self._in_flight_orders.values():
+            typed_in_flight_order = in_flight_order
+            if ((typed_in_flight_order.order_type is not OrderType.LIMIT) or
+                    typed_in_flight_order.is_done):
+                continue
+            retval.append(typed_in_flight_order.to_limit_order())
+
+        return retval
+
+    @property
+    def expiring_orders(self) -> List[LimitOrder]:
+        return [self._in_flight_orders[order_id].to_limit_order()
+                for _, order_id
+                in self._order_expiry_queue]
+
+    @property
+    def tracking_states(self) -> Dict[str, any]:
+        return {
+            key: value.to_json()
+            for key, value in self._in_flight_orders.items()
+        }
+
+    def restore_tracking_states(self, saved_states: Dict[str, any]):
+        self._in_flight_orders.update({
+            key: InFlightOrder.from_json(value)
+            for key, value in saved_states.items()
+        })
+
+    async def get_active_exchange_markets(self):
+        return await DolomiteAPIOrderBookDataSource.get_active_exchange_markets()
+
+    async def _status_polling_loop(self):
+        while True:
+            try:
+                self._poll_notifier = asyncio.Event()
+                await self._poll_notifier.wait()
+
+                self._update_balances()
+                await asyncio.gather(
+                    self._update_trading_rules(),
+                    self._update_order_status(),
+                    self._update_trade_fees()
+                )
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().network(
+                    "Unexpected error while fetching account and status updates.",
+                    exc_info=True,
+                    app_warning_msg=f"Failed to fetch account updates on Dolomite. Check network connection."
+                )
+
+    def _update_balances(self):
+        self._account_balances = self.wallet.get_all_balances()
+
+        
+    async def _update_trading_rules(self):
+        cdef:
+            double current_timestamp = self._current_timestamp
+
+        if current_timestamp - self._last_update_trading_rules_timestamp > 60.0 or len(self._trading_rules) < 1:
+            
+            markets = await self.list_market()
+            
+            info_url = f"{self.DOLOMITE_REST_ENDPOINT}/v1/info"
+            res = await self._api_request(http_method="get", url=info_url)
+            
+            min_maker_order = float(res["data"]["min_usd_maker_trade_amount"]["amount"]) / math.pow(10, 5)
+            min_taker_order = float(res["data"]["min_usd_taker_trade_amount"]["amount"]) / math.pow(10, 5)
+            
+            
+            markets = await self.get_market("WETH-DAI")
+            weth_price = float(market["data"]["current_price"]["amount"]) / math.pow(10, 18)
```

Comment:
It's ok to hardcode 18 here, but be careful.

Perhaps you should define 18 as a constant for WETH and DAI.
IE `WETH_PRECISION = 18` and `DAI_PRECISION = 18`